### ISSUE TITLE
KTX2 support for threejs.com/editor

### ISSUE THREAD

**Issue by hybridherbst** [mrdoob/three.js#22035]
**Is your feature request related to a problem? Please describe.**

Threejs supports KTX2 when loading GLTF files, but the editor doesn't seem to support it yet (the KTX2 loader isn't registered according to the console).

**Describe the solution you'd like**

GLTF files that use KTX2 texture compression should open in threejs editor.

**Additional context**

![image](https://user-images.githubusercontent.com/2693840/122990317-1aa43b00-d3a4-11eb-8d0d-d3bf1dffdb60.png)

**Comment by donmccurdy** [comment]
Would loading GLTF w/ KTX2 textures still be useful to you if the Editor can't export the scene after displaying it? Unfortunately that is a rather more complicated thing to fix, since the we can't reconstruct the original KTX2 texture from the transcoded data. Some discussion in https://github.com/mrdoob/three.js/issues/13911#issuecomment-616492588.

**Comment by hybridherbst** [comment]
Ah, I see. What I was _actually_ trying is indeed to export to USDZ, as I was investigating https://github.com/google/model-viewer/issues/2503 (besides other issues, model-viewer auto-export to USDZ not working when model has ktx2 textures). Seems that's kind of the same issue then – as the textures can't be read back to CPU they can't be converted.

I'm not sure I quite understand the issue though, couldn't the KTX-unpacked texture be blitted into an ARGB32 target and read back to the CPU for conversion to any other format? (that's what the flow would be in e.g. Unity)

**Comment by donmccurdy** [comment]
Ah, probably same issue with model-viewer then, yes.

Rendering to RGBA target would be a good fallback. Perhaps lossy depending on the device where it's running... e.g. if we've already transcoded to PVRTC on an iOS device, KTX2 -> PVRTC -> RGBA (for USDZ) will be more lossy than KTX2 -> RGBA, I think. The proposal in #13911 was to add a new TranscodableTexture class that retains the pre-transcode data in some form.

**Comment by hybridherbst** [comment]
I might be mistaken but I think KTX2 (usually ETC1S or UASTC) is already lossy, so not sure if there's actually a noticeable quality difference between these paths. Should I keep this issue here open?

**Comment by donmccurdy** [comment]
Right, KTX2 is lossy compression. Transcoding KTX (i.e. ETC1S) to older formats like PVRTC (as compared to ETC1/2, preferred) may lose a bit more. This isn't a blocker on the approach, just something to be aware of. I'm not sure what solution others would prefer (editor converts textures? add this feature to USDZExporter? other exporters?) so probably good to keep this open for discussion.

**Comment by Mugen87** [comment]
Since the editor should now display KTX2 textures (embedded in glTF assets as well as selected via the UI), this issue can be closed. 

The missing serialization/deserialization support is tracked in #17974. #13911 also discusses a possible solution based on `TranscodableTexture`.
